### PR TITLE
sGAR ~ 2024.06.20 ~ minor bugs for the stuff.

### PR DESCRIPTION
**Pull request by ame-test**
**What changes does this PR bring?**
* https://collaboration.homeoffice.gov.uk/jira/browse/FAIR-528

Basically, if you submit the gar without a value in the IATA/ICAO section and with an invalid data, it sued ot default to teh longtitude section, now it defaults to the latitude section.

## Previous bug
[
Screenshot 2024-06-20 at 09 53 38
Screenshot 2024-06-20 at 09 53 33
](url)

## Fixed
Screenshot 2024-06-20 at 12 08 57

** Tests **

image

**Definition of Done**
The following need to be checked off as they are complete as part of the
PR process. If you are unable to mark off an item as complete, then state
in the comments why this is the case.
- [ ] Have you built the code locally and confirmed its working?
- [ ] Is the build confirmed to be passing on CI?
- [ ] Have you added enough relevant unit tests for your change?
- [ ] Have you added enough relevant E2E tests for your change?
- [ ] Have you updated any relevant documentation as part of this change?
- [ ] Has this change been tested against the Acceptance Criteria?
- [ ] Have you considered how this will be deployed in SIT/Staging/Production?

### CODE REVIEW

File: src/common/models/Cookie.class.js [hunk 1]
```diff
@@ -245,7 +247,9 @@ class Cookie {
     this.session.gar.voyageDeparture.departurePort = voyageObj.departurePort;
     this.session.gar.voyageDeparture.departureLat = trimToDecimalPlaces(voyageObj.departureLat, 6);
     this.session.gar.voyageDeparture.departureLong = trimToDecimalPlaces(voyageObj.departureLong, 6);
-    this.session.gar.voyageDeparture.departurePortChoice = voyageObj.portChoice || (voyageObj.departureLat || voyageObj.departureLong) ? 'No' : 'Yes';
+
```
Comment (ninaknowlesHO) [inline src/common/models/Cookie.class.js:250]: you could move some of the logic from the getters and setters into a helper method if you want to, think this is outside the scope of the ticket though

**Review by ninaknowlesHO — APPROVED** [review]

File: src/common/models/Cookie.class.js (hunk 1 above)
Comment (ame-test) [reply src/common/models/Cookie.class.js:231]: Has passed e2e tests (with a different version to handle other bugs).
* A mannual test found it is ok.

File: src/common/models/Cookie.class.js [hunk 2]
```diff
@@ -766,6 +756,32 @@ class Cookie {
     return `${h}:${m}`;
   }
 
+  parseVoyageObj(type, voyageObj) {
+    let voyage = {};
```
Comment (ninaknowlesHO) [inline src/common/models/Cookie.class.js:760]: Use const not let as it is not being reassigned

File: src/common/models/Cookie.class.js [hunk 3]
```diff
@@ -766,6 +756,32 @@ class Cookie {
...
+    if (!voyageObj[`${type}Date`]) {
```
Comment (ninaknowlesHO) [inline src/common/models/Cookie.class.js:761]: This is a bit clearer I think:

`    voyage[`${type}Date`] = voyageObj[`${type}Date`] || this.generateDate(
        voyageObj[`${type}Day`],
        voyageObj[`${type}Month`],
        voyageObj[`${type}Year`]
    );

    voyage[`${type}Time`] = voyageObj[`${type}Time`] || this.generateTime(
        voyageObj[`${type}Hour`],
        voyageObj[`${type}Minute`]
    );`

File: src/common/models/Cookie.class.js [hunk 4]
```diff
@@ -766,6 +756,32 @@ class Cookie {
...
+    } else {
+      // Set voyage from API response, so dates and times are already built
+      voyage[`${type}Date`] = voyageObj[`${type}Date`];
+      voyage[`${type}Time`] = voyageObj[`${type}Time`];
+    }
+    voyage[`${type}Port`] = voyageObj[`${type}Port`];
+    voyage[`${type}Long`] = trimToDecimalPlaces(voyageObj[`${type}Long`], 6);
+    voyage[`${type}Lat`] = trimToDecimalPlaces(voyageObj[`${type}Lat`], 6);
+    
+    const defaultPortChoice = (voyageObj[`${type}Lat`] || voyageObj[`${type}Long`]) ? 'No' : 'Yes';
```
Comment (ninaknowlesHO) [inline src/common/models/Cookie.class.js:779]: Consider splitting the logic here to improve readability, e.g. 

```
 const hasCoordinates = voyageObj[`${type}Lat`] || voyageObj[`${type}Long`];
    const defaultPortChoice = hasCoordinates ? 'No' : 'Yes';
```

Comment (ame-test) [reply src/common/models/Cookie.class.js:779]: I don't think it matters enough to merit changing.
* One can argue that it is being over split at this point, making it harder to follow.
* In truth, it is kinda splitting hairs.
* If I am wrong let me know.

File: src/common/models/Cookie.class.js (hunk 2 above)
Comment (ame-test) [reply src/common/models/Cookie.class.js]: Sure, it is done.

 I guess when I was originally doing it, I though that using a `const` for a variable whose entire purpose to mutated seem a bit a weird, but I am also happy for the changes.

File: src/common/models/Cookie.class.js (hunk 3 above)
Comment (ame-test) [reply src/common/models/Cookie.class.js]: This changes has been made.

**Review by ninaknowlesHO — APPROVED** [review]
